### PR TITLE
Ntopng config UI changes to allow arbitrary local networks list

### PR DESCRIPTION
**Pull request by madpilot78**
ntopng allows an arbitrary list of networks to be considered as "local".

I've added ability to configure such an arbitrary list to the pfsense ntopng UI.

To accommodate for such options I have reorganised a little the UI layout.

I'm submitting it hoping it can be accepted in your official tree. If any changes in UI layout or functionality is required please comment on this pull request.

Thanks!

### CODE REVIEW

File: net/pfSense-pkg-ntopng/files/usr/local/pkg/ntopng.inc [hunk 1]
```diff
@@ -121,6 +121,17 @@ function ntopng_sync_package() {
 				$local_networks = "--local-networks " . escapeshellarg(gen_subnet(get_interface_ip('lan'), get_interface_subnet('lan')) . '/' . get_interface_subnet('lan'));
 			}
 			break;
+		case "custom":
+			$nets = array();
+			foreach ($ntopng_config['row'] as $net) {
+				if (is_subnet($net['cidr'])) {
+					$nets[] = $net['cidr'];
```
Comment (jim-p) [inline net/pfSense-pkg-ntopng/files/usr/local/pkg/ntopng.inc]: To be on the safe side, you should use `trim()` around the right side, to be absolutely certain there is no whitespace around any of the values.

Comment (madpilot78) [inline net/pfSense-pkg-ntopng/files/usr/local/pkg/ntopng.inc]: You're right. I should have thought about this right away.

Fixing it shortly.

**Comment by madpilot78** [comment]
I noticed the pfSense-pkg-ntopng package version has already been bumped to 0.8.7.

Should I add a commit bumping it further to 0.8.8?

**Comment by jim-p** [comment]
Yes, you'll want to rebase and then bump it again.

**Comment by madpilot78** [comment]
Oops, I just wanted to rebase and bump portrevision but made a git mess somehow.

Not sure how to fix it.

Any suggestions?

**Comment by doktornotor** [comment]
Perhaps someone could commit this before it gets messed up again? 😛

**Comment by dennypage** [comment]
Guys, there's a couple of bugs here. Even though the custom networks option is not selected, the custom networks input field is still shown. More significantly, the custom networks input field is checked for validity even if the custom networks option is not selected. An empty list is not considered valid. This means that everyone's prior config from 0.8.8 is considered invalid and they cannot re-save a configuration without entering a valid cidr.

**Comment by dennypage** [comment]
And yes, putting a valid CIDR in the field does allow the config to be saved. I've only tested a couple, but a common one, "192.168.0.0/16" works fine.

**Comment by madpilot78** [comment]
Just for the record, I have submitted pull request #372 which contains a fix for this issue.

**Pull request merged**